### DISCUSSION TITLE
AI summaries or structured data suggestions for note frontmatter

### DISCUSSION THREAD

**Discussion by nweii** [brianpetro/obsidian-smart-connections#453] [Ideas]
I saw [this tweet](https://twitter.com/thesephist/status/1754150819451248744) and wondered if we could have something similar in Obsidian through this plugin's embeddings and models:
- Generate a note summary of a user-specified length for a "Summary" frontmatter item
- Suggest new and existing tags for the note
- Suggest other frontmatter keys that might apply to the current note
- Suggest aliases

**Comment by brianpetro** [comment]
Hey @nweii 

I saw and thought the same 😂

My next thought was maybe front matter isn't ideal for longer blocks of text like summaries, but I don't use front matter too much, so I can't say for sure.

Tag suggestions are something that's been requested before, and I think now might be a good time for me too look into that a bit more. While it's easy to get GPT to suggest tags, my requirement for Smart Connections is that those tags be aligned with other tags already in the vault. For example, if three tags are suggested, no more than one of those should be a new tag. 

Front matter suggestions, on the backend, look really similar to the tag suggestions, with similar requirements.

I haven't thought about the alias generating, probably because I have like one or two aliases in my entire vault. I imagine it would be quite similar to the tags and front matter generation, though.

🌴

**Comment by RealRavens** [comment]
Very interesting. Without considering any UI redesigns that may be happening, Smart Connections could have a 3rd panel for the open note and other features. Sorta like how properties in obsidian has the properties panel.  Especially with @brianpetro adding a more vault aware tool to the AI soon. So when you enter that panel you can just get suggestions and click to add them in to your frontmatter. Since the formatting for properties is consistent, it's not too much a hastle to get templating right. Obviously this would be quite a brute force approach design wise. It could be a good m.v.p to see it implemented and go from there. Cool regardless 👍
